Components/LearningCore/Source/gov/sandia/cognition/learning/algorithm/clustering/SNNClusterer.java

```diff
@@ -0,0 +1,637 @@
+    {
+        List<DataType> result = new ArrayList<>();
+        Set<DataType> neighbors = this.knnMap.get(point);
+
+        // Loop through other points to look for mutual neighbors
+        for (DataType otherPoint : this.points)
+        {
+            if (otherPoint != point)
+            {
+                Set<DataType> otherNeighbors = this.knnMap.get(otherPoint);
+                Set<DataType> sharedNeighbors = new HashSet<>(neighbors);
+                sharedNeighbors.retainAll(otherNeighbors);
+
+                // If more than eps neighbors, add it to results
+                if (sharedNeighbors.size() >= this.eps)
+                {
+                    result.add(otherPoint);
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Takes a core point and adds its neighbors to the cluster. Also merges any
+     * neighboring clusters by iterating through neighbors and seeing if they
+     * are core points. Records any processed points as "clustered".
+     *
+     * @param point The core point to expand the cluster around
+     * @param neighbors The core point's neighbors.
+     */
+    private void expandCluster(DataType point,
+        List<DataType> neighbors)
+    {
+        LinkedList<DataType> neighborsLinkedList = new LinkedList<>(neighbors);
+        this.currentCluster.add(point);
+        this.clustered.add(point);
+
+        // Loop through each neighbor and process it
+        while (!neighborsLinkedList.isEmpty())
+        {
+            DataType neighbor = neighborsLinkedList.remove();
+
+            if (!this.visited.contains(neighbor))
+            {
+                this.visited.add(neighbor);
+                List<DataType> neighborNeighbors = this.getConnectedNeighbors(
+                    neighbor);
+
+                // If neighbor is a core point, process its neighbors too
+                if (neighborNeighbors.size() >= this.minPts)
+                {
+                    neighborsLinkedList.addAll(neighborNeighbors);
+                }
+            }
+            // Mark neighbors as clustered and add them to current cluster
+            if (!this.clustered.contains(neighbor))
+            {
+                this.currentCluster.add(neighbor);
+                this.clustered.add(neighbor);
+            }
+        }
+    }
+
+    /**
+     * Adds noise cluster to list of clusters.
+     */
+    @Override
+    protected void cleanupAlgorithm()
+    {
+        this.clusters.set(0, this.creator.createCluster(this.noiseCluster));
+    }
+
+    /**
+     * Gets a list of the clusters.
+     *
+     * @return a list of the clusters
+     */
+    @Override
+    public Collection<ClusterType> getResult()
+    {
+        return this.getClusters();
+    }
+
+    /**
+     * Sets the number of nearest neighbors to use.
+     *
+     * @param k The number of nearest neighbors k.
+     */
+    public void setNumNeighbors(int k)

```

Comment:
Minor naming: Normally these setters/getters have a one-to-one mapping to the fields. So maybe rename "k" to numNeighbors?